scripts/validate_metadata/validate_metadata.py

```diff
@@ -0,0 +1,319 @@
+
+        elif field == "tags":
+            tags_val = metadata.get("tags")
+            if not (isinstance(tags_val, list)):
+                raise ValidationError("{} value identified in field 'tags' in {} for component '{}'. Value must be string array.".format(value_type, METADATA, component_name))
+            if not all(isinstance(item, str) for item in tags_val):
+                raise ValidationError("The following tags in {} for component '{}': {}. Expected an array of scalar strings.".format(METADATA, component_name, tags_val))
+        elif field == "ci":
+            ci_val = metadata.get("ci")
+            if not isinstance(ci_val, dict):
+                raise ValidationError("{} value identified for field 'ci' in {} for component '{}'. Value must be dictionary.".format(value_type, METADATA, component_name))
+            keys = set(ci_val.keys())
+            if not (keys == {"skip_dependency_probe"}):
+                raise ValidationError("The following field(s) were found in field 'ci' in {} for component '{}': {}. Only field 'skip_dependency_probe' is valid.".format(METADATA, component_name, list(ci_val.keys())))
+            probe = ci_val.get("skip_dependency_probe")
+            if probe is not None and not isinstance(probe, bool):
+                raise ValidationError("{} expects a boolean value for skip_dependency_probe but {} value provided: '{}'.".format(METADATA, type(probe).__name__, probe))
+
+        elif field == "links":
+            links_value = metadata.get("links")
+            if not isinstance(links_value, dict):
+                raise ValidationError("{} value identified in field 'links' in {} for component '{}'. Value must be dictionary.".format(value_type, METADATA, component_name))
+
+def get_invalid_versions(dependencies: list[dict]) -> list[dict]:
+    """Return a list of the input dependencies that contain invalid semantic versioning.
+
+    Args:
+        dependencies: list[dict] of dependencies to be validated
+
+    Return:
+        dependencies: list[dict] of invalid dependencies
+    """
+    if dependencies is None:
+        return []
+    invalid : list[dict] = []
+    for dependency in dependencies:
+        version = dependency.get("version")
+        # If the dependency version is null or non-string, it is invalid.
+        if version is None or not isinstance(version, str):
+            invalid.append(dependency)
+        # Strip leading '==', '>=' or '<=' from dependency version, if applicable.
+        if len(version) > 1 and version[:2] in COMPARISON:
+            version = version[2:]
+        if not Version.is_valid(version):
+            invalid.append(dependency)
+    return invalid
+
+def main():
+    """Main entry point for the CLI."""
+    args = parse_args()
+    if not args.component:
+        raise ValidationError("Error: --component must be specified.")
+
```

Comment:
the nice thing about arg parse is that you no longer need to do self validation on flags, you can just add a requirement on the `--component` flag like this: 

```python
    parser.add_argument(
        '--component',
        type=validate_component_directory,
        required=True,  # <----- here
        help='Path to the component directory (must contain component.py and metadata.yaml)'
    )
```

so then you get useful commands automatically like: 

```bash
*# python scripts/validate_metadata/validate_metadata.py 
usage: validate_metadata.py [-h] --component COMPONENT
validate_metadata.py: error: the following arguments are required: --component
```